koku/masu/util/ocp/common.py

```diff
@@ -199,9 +200,15 @@ def get_report_details(report_directory):
                 payload_dict["date"] = parser.parse(payload_dict["date"])
                 payload_dict["manifest_path"] = manifest_path
                 # parse start and end dates if in manifest
-                for field in ["start", "end"]:
-                    if payload_dict.get(field):
-                        payload_dict[field] = parser.parse(payload_dict[field])
+                if payload_dict.get["start"]:
+                    start = payload_dict.get["start"]
+                    payload_dict["start"] = parser.parse(start)
+                if start and payload_dict.get["end"]:
+                    today = dh.today
+                    end = payload_dict.get["end"]
+                    if start.year == today.year and start.month == today.month and (end.month != today.month):
```

Comment:
```suggestion
                    # We override the end date from the first of the next month to the end of current month
                    # We do this to prevent summary from triggering unnecessarily on the next month
                    if start.year == today.year and start.month == today.month and (end.month != today.month):
```